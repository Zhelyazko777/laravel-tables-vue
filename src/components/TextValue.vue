<template>
  <span 
    :title="showTooltipOnHover  ? tooltip : ''"
    :data-toggle="showTooltipOnHover  ? 'tooltip' : ''"
  >
    <span class="prefix">{{prefix}}</span>
    <span class="value">{{value}}</span>
    <span class="suffix">{{suffix}}</span>
  </span>
</template>

<script>
export default {
  props: {
    value: {
      required: true,
      type: String|null,
    },
    prefix: {
      required: true,
      type: String|null,
    },
    suffix: {
      required: true,
      type: String|null,
    },
    tooltip: {
      required: true,
      type: String|null,
    },
    showTooltipOnHover: {
      required: true,
      type: Boolean|null,
    },
  },
  computed: {
    cellValue () {
      let result = '-';

      if (this.value) {
        result = this.value;
      }
      if (this.prefix) {
        result = this.prefix + result;
      }
      if (this.suffix) {
        result = result + this.suffix;
      }

      return result;
    }
  }
}
</script>