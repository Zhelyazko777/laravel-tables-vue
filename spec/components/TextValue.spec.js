import { mount } from '@vue/test-utils'
import TextValue from './../../src/components/TextValue.vue';

describe('TextValue.vue Test', () => {
  it('Should render corrrect value and empty suffix and preffix', () => {
    const wrapper = mount(TextValue, {
      propsData: {
        value: 'test value',
        prefix: null,
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();      
  });

  it('Should render corrrect value and prefix if passed and no suffix', () => {
    const wrapper = mount(TextValue, {
      propsData: {
        value: 'test value',
        prefix: 'prefix',
        suffix: null,
        tooltip: null,
        showTooltipOnHover: false,
      }
    });

    expect(wrapper).toMatchSnapshot();      
  });

  it('Should render corrrect value, suffix and prefix if passed', () => {
    const wrapper = mount(TextValue, {
      propsData: {
        value: 'test value',
        prefix: 'prefix',
        suffix: 'suffix',
        tooltip: null,
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should render correct tooltip if passed and showTooltipOnHover is true', () => {
    const wrapper = mount(TextValue, {
      propsData: {
        value: null,
        prefix: null,
        suffix: null,
        tooltip: 'Some tooltip',
        showTooltipOnHover: true,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });

  it('Should not render correct tooltip if passed and showTooltipOnHover is false', () => {
    const wrapper = mount(TextValue, {
      propsData: {
        value: null,
        prefix: null,
        suffix: null,
        tooltip: 'Some tooltip',
        showTooltipOnHover: false,
      }
    });
 
    expect(wrapper).toMatchSnapshot();        
  });
});