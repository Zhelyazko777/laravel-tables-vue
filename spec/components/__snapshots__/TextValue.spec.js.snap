// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextValue.vue Test Should not render correct tooltip if passed and showTooltipOnHover is false 1`] = `<span title="" data-toggle=""><span class="prefix"></span> <span class="value"></span> <span class="suffix"></span></span>`;

exports[`TextValue.vue Test Should render correct tooltip if passed and showTooltipOnHover is true 1`] = `<span title="Some tooltip" data-toggle="tooltip"><span class="prefix"></span> <span class="value"></span> <span class="suffix"></span></span>`;

exports[`TextValue.vue Test Should render corrrect value and empty suffix and preffix 1`] = `<span title="" data-toggle=""><span class="prefix"></span> <span class="value">test value</span> <span class="suffix"></span></span>`;

exports[`TextValue.vue Test Should render corrrect value and prefix if passed and no suffix 1`] = `<span title="" data-toggle=""><span class="prefix">prefix</span> <span class="value">test value</span> <span class="suffix"></span></span>`;

exports[`TextValue.vue Test Should render corrrect value, suffix and prefix if passed 1`] = `<span title="" data-toggle=""><span class="prefix">prefix</span> <span class="value">test value</span> <span class="suffix">suffix</span></span>`;
